src/plugins/Wallet/wallet.ts

```diff
@@ -57,21 +57,22 @@ export async function getManagedWallets(): Promise<{
             .map(WalletRecordOutDB)
             .filter((x): x is ManagedWalletRecord => x.type !== 'exotic')
             .map(async (record) => ({ ...record, privateKey: await makePrivateKey(record) }))
-        const recordWithKeys = await Promise.all(records)
-        return recordWithKeys.sort((a, b) => {
-            if (a._wallet_is_default) return -1
-            if (b._wallet_is_default) return 1
-            if (a.updatedAt > b.updatedAt) return -1
-            if (a.updatedAt < b.updatedAt) return 1
-            if (a.createdAt > b.createdAt) return -1
-            if (a.createdAt < b.createdAt) return 1
-            return 0
-        })
+        return (await Promise.all(records))
+            .filter((record) => record.privateKey) // filter out legacy DB which has got invalid private key
```

Comment:
maybe we should not just hide those records. that cause wallet lost

Reply:
I think you are right.